aptos-move/aptos-vm/src/move_vm_ext/write_op_converter.rs

```diff
@@ -161,6 +166,33 @@ impl<'r> WriteOpConverter<'r> {
         }
     }
 
+    /// Converts module bytes and their compiled representation extracted from publish
+    /// request into write ops. Only used by V2 loader implementation.
+    pub(crate) fn convert_modules_into_write_ops<'a>(
+        &self,
+        module_storage: &impl AptosModuleStorage,
+        code_and_modules: impl IntoIterator<Item = (Bytes, &'a CompiledModule)>,
+    ) -> PartialVMResult<BTreeMap<StateKey, WriteOp>> {
+        let mut write_ops = BTreeMap::new();
+        for (bytes, compiled_module) in code_and_modules.into_iter() {
+            let addr = compiled_module.self_addr();
+            let name = compiled_module.self_name();
+            let state_key = StateKey::module(addr, name);
+
+            let module_exists = module_storage.check_module_exists(addr, name)?;
+            let op = if module_exists {
+                Op::Modify(bytes)
+            } else {
+                Op::New(bytes)
+            };
+
+            // TODO(loader_v2): Query state value metadata from module storage.
```

Comment:
What does this todo mean?

Reply:
`AptosModuleStorage` should have `fetch_module_state_value_metadata(...)`

Reply:
See #14341